tests/phpunit/CRM/Civicase/Service/CaseCategoryFromUrlTest.php

```diff
@@ -0,0 +1,398 @@
+
+    $this->assertEquals($category['name'], $categoryName);
+  }
+
+  /**
+   * Test get category name from category Id in request.
+   *
+   * Test the detection of the category name, when it is received the category
+   * Id on the request body.
+   */
+  public function testGetCategoryNameFromRequestBodyContainingTheCategoryId() {
+    $category = CaseCategoryFabricator::fabricate();
+    $categoryFromUrlServiceMock = $this->mockRequestResponse([$category['value']]);
+    $requestedUrl = $this->getUrlByType(CategoryFromUrlService::CASE_CATEGORY_TYPE_URL);
+
+    $categoryName = $categoryFromUrlServiceMock->get($requestedUrl);
+
+    $this->assertEquals($category['name'], $categoryName);
+  }
+
+  /**
+   * Test get category name from query string.
+   *
+   * Test the detection of the category name, when it is received that name
+   * on the query string of the URL.
+   */
+  public function testGetCategoryNameFromUrlQueryStringContainingTheCategoryName() {
+    $category = CaseCategoryFabricator::fabricate();
+
+    $requestedUrl = $this->getUrlByType(CategoryFromUrlService::CASE_CATEGORY_TYPE_URL);
+    $requestedUrlWithQueryString = $this->getUrlWithQueryString($requestedUrl, $category['name']);
+
+    $categoryFromUrlServiceMock = $this->mockRequestResponse(
+      [NULL, $requestedUrlWithQueryString]
+    );
+
+    $categoryName = $categoryFromUrlServiceMock->get($requestedUrl);
+
+    $this->assertEquals($category['name'], $categoryName);
+  }
+
+  /**
+   * Test get category name from activity Id in request.
+   *
+   * Test the detection of the category name, when it is received an activity
+   * Id on the request body.
+   */
+  public function testGetCategoryNameFromActivityIdInRequestBody() {
+    $case = $this->createCase();
+    $activity = $this->createActivityForCaseId($case['id']);
+    $categoryFromUrlServiceMock = $this->mockRequestResponse([$activity['id']]);
+    $requestedUrl = $this->getUrlByType(CategoryFromUrlService::ACTIVITY_TYPE_URL);
+
+    $categoryName = $categoryFromUrlServiceMock->get($requestedUrl);
+
+    $this->assertEquals(
+      $this->getCaseTypeCategoryNameByCaseTypeId($case['case_type_id']),
+      $categoryName
+    );
+  }
+
+  /**
+   * Test get category name from invalid activity Id in request.
+   *
+   * Test that NULL is return from the detection of the category name,
+   * when it is received an invalid activity Id on the request body.
+   */
+  public function testGetCategoryNameFromInvalidActivityIdInRequestBody() {
+    $categoryFromUrlServiceMock = $this->mockRequestResponse([rand()]);
+    $requestedUrl = $this->getUrlByType(CategoryFromUrlService::ACTIVITY_TYPE_URL);
+
+    $categoryName = $categoryFromUrlServiceMock->get($requestedUrl);
+
+    $this->assertNull($categoryName);
+  }
+
+  /**
+   * Test get category name from category Id in case entity ajax request.
+   *
+   * Test the detection of the category name, when it is received the category
```

Comment:
s/when it is received the category/when it is received from the category

Reply:
Done here https://github.com/compucorp/uk.co.compucorp.civicase/pull/663/commits/cd2dcd86d5d7fa5fad879f7a57761e0b27e4e531 (I committed the text changes with the other modification on tests)